src/core/thread/mesh_forwarder.cpp

```diff
@@ -1957,12 +1963,12 @@ void MeshForwarder::HandleDataRequest(const Mac::Address &aMacSource, const Thre
     VerifyOrExit(mNetif.GetMle().GetDeviceState() != Mle::kDeviceStateDetached);
 
     VerifyOrExit((child = mNetif.GetMle().GetChild(aMacSource)) != NULL);
-    child->mLastHeard = Timer::GetNow();
-    child->mLinkFailures = 0;
+    child->SetLastHeard(Timer::GetNow());
+    child->ResetLinkFailures();
 
-    if (!mSourceMatchController.IsEnabled() || (child->GetQueuedMessageCount() > 0))
+    if (!mSourceMatchController.IsEnabled() || (child->GetIndirectMessageCount() > 0))
     {
-        child->mDataRequest = true;
+        child->SetDataRequestPending(true);
```

Comment:
`SetDataRequestPending()` is to record the fact that an IEEE 802.15.4 Data Request message was received from the child and needs to be serviced.